### PR TITLE
Add rak5010 to HIL testing

### PR DESCRIPTION
**Pull request by szczys**
Add support for the RAK5010 in HIL testing.

resolves https://github.com/golioth/firmware-issue-tracker/issues/530

### CODE REVIEW

File: .github/workflows/hil_tests.yml [hunk 1]
```diff
@@ -110,22 +110,26 @@ jobs:
...
+            extra_twister_args: '--west-flash="--dev-id=${!SNR_VAR}"'
             run_tests: true
 
           - hil_board: nrf9160dk
             west_board: nrf9160dk_nrf9160_ns
             manifest: .ci-west-ncs.yml
+            extra_twister_args: '--west-flash="--dev-id=${!SNR_VAR}"'
             run_tests: true
 
           - hil_board: rak5010_nrf52840
```
Comment (szczys) [inline .github/workflows/hil_tests.yml:128]:
```suggestion
          - hil_board: rak5010
```

Use shorted board name. This matches the self-hosted runner board recognition script

File: .github/workflows/hil_sample_zephyr.yml [hunk 2]
```diff
@@ -226,6 +241,17 @@ jobs:
           elif [[ "${hil_board}" == "mimxrt1024_evk" ]]; then
             JLinkExe -nogui 1 -if swd -speed auto -device MIMXRT1024XXX5A -CommanderScript erase_mimxrt1024_evk.jlink -SelectEmuBySN ${!SNR_VAR}
 
+          elif [[ "${hil_board}" == "rak5010_nrf52840" ]]; then
+            gdb \
+            -ex "set confirm off" \
+            -ex "target extended-remote "${BLACKMAGIC_PORT} \
```
Comment (szczys) [inline .github/workflows/hil_sample_zephyr.yml:247]:
```suggestion
            -ex "target extended-remote "${CI_BLACKMAGICPROBE_PORT} \
```

Update to match what self-hosted runner board recogntiion script generates.

File: examples/zephyr/fw_update/sample.yaml [hunk 3]
```diff
@@ -10,10 +10,8 @@ tests:
     timeout: 600
     extra_args: EXTRA_CONF_FILE="../common/runtime_settings.conf"
     platform_allow: >
-      esp32_devkitc_wrover
```
Comment (sam-golioth) [inline examples/zephyr/fw_update/sample.yaml:13]: Is this change intentional?

File: .github/workflows/hil_sample_zephyr.yml [hunk 4]
```diff
@@ -197,7 +197,8 @@ jobs:
                 --pytest-args="--api-key=${{ secrets[inputs.api-key-id] }}"                       \
                 --pytest-args="--wifi-ssid=${{ secrets[format('{0}_WIFI_SSID', runner.name)] }}"  \
                 --pytest-args="--wifi-psk=${{ secrets[format('{0}_WIFI_PSK', runner.name)] }}"    \
-                --pytest-args="--mask-secrets"
+                --pytest-args="--mask-secrets"                                                    \
+                -v
```
Comment (sam-golioth) [inline .github/workflows/hil_sample_zephyr.yml:201]: 🙌

**Review by sam-golioth — APPROVED** [review]
Nice work @szczys!

File: examples/zephyr/fw_update/sample.yaml (hunk 3 above)
Comment (szczys) [reply examples/zephyr/fw_update/sample.yaml:13]: it was intentional to move this board to the build only. But as you've rightly identified, that's outside the scope of this PR so I'll remove this change before merging. Thanks!
